worlds/stardew_valley/options.py

```diff
@@ -553,56 +533,78 @@ class Mods(OptionSet):
-    EntranceRandomization,
-    SeasonRandomization,
-    Cropsanity,
-    BackpackProgression,
-    ToolProgression,
-    SkillProgression,
-    BuildingProgression,
-    FestivalLocations,
-    ElevatorProgression,
-    ArcadeMachineLocations,
-    SpecialOrderLocations,
-    HelpWantedLocations,
-    Fishsanity,
-    Museumsanity,
-    Friendsanity,
-    FriendsanityHeartSize,
-    NumberOfMovementBuffs,
-    NumberOfLuckBuffs,
-    ExcludeGingerIsland,
-    TrapItems,
-    MultipleDaySleepEnabled,
-    MultipleDaySleepCost,
-    ExperienceMultiplier,
-    FriendshipMultiplier,
-    DebrisMultiplier,
-    QuickStart,
-    Gifting,
-    Mods,
-]
-stardew_valley_options: Dict[str, type(Option)] = {option.internal_name: option for option in
-                                                   stardew_valley_option_classes}
-default_options = {option.internal_name: option.default for option in stardew_valley_options.values()}
-stardew_valley_options["death_link"] = DeathLink
-
-
-def fetch_options(world, player: int) -> StardewOptions:
-    return StardewOptions({option: get_option_value(world, player, option) for option in stardew_valley_options})
-
-
-def get_option_value(world, player: int, name: str) -> Union[bool, int]:
-    assert name in stardew_valley_options, f"{name} is not a valid option for Stardew Valley."
-
-    value = getattr(world, name)
-
-    if issubclass(stardew_valley_options[name], Toggle):
-        return bool(value[player].value)
-    return value[player].value
+@dataclass
+class StardewValleyOptions(PerGameCommonOptions):
+    goal: Goal
+    starting_money: StartingMoney
+    profit_margin: ProfitMargin
+    bundle_randomization: BundleRandomization
+    bundle_price: BundlePrice
+    entrance_randomization: EntranceRandomization
+    season_randomization: SeasonRandomization
+    cropsanity: Cropsanity
+    backpack_progression: BackpackProgression
+    tool_progression: ToolProgression
+    skill_progression: SkillProgression
+    building_progression: BuildingProgression
+    festival_locations: FestivalLocations
+    elevator_progression: ElevatorProgression
+    arcade_machine_locations: ArcadeMachineLocations
+    special_order_locations: SpecialOrderLocations
+    help_wanted_locations: HelpWantedLocations
+    fishsanity: Fishsanity
+    museumsanity: Museumsanity
+    friendsanity: Friendsanity
+    friendsanity_heart_size: FriendsanityHeartSize
+    number_of_movement_buffs: NumberOfMovementBuffs
+    number_of_luck_buffs: NumberOfLuckBuffs
+    exclude_ginger_island: ExcludeGingerIsland
+    trap_items: TrapItems
+    multiple_day_sleep_enabled: MultipleDaySleepEnabled
+    multiple_day_sleep_cost: MultipleDaySleepCost
+    experience_multiplier: ExperienceMultiplier
+    friendship_multiplier: FriendshipMultiplier
+    debris_multiplier: DebrisMultiplier
+    quick_start: QuickStart
+    gifting: Gifting
+    mods: Mods
+    death_link: DeathLink
+
+    def get_option_value(self, name: str) -> int:
```

Comment:
see above comment. this is literally code duplication lol. I'll see about finding a good spot to document it.